impl/src/main/java/io/jsonwebtoken/impl/DefaultJwtBuilder.java

```diff
@@ -693,14 +687,7 @@ private String encrypt(final Payload content, final Key key, final Provider keyP
         Assert.stateNotNull(keyAlgFunction, "KeyAlgorithm function cannot be null.");
         assertPayloadEncoding("JWE");
 
-        InputStream plaintext;
-        if (content.isClaims()) {
-            ByteArrayOutputStream out = new ByteArrayOutputStream(4096);
-            writeAndClose("JWE Claims", content, out);
-            plaintext = Streams.of(out.toByteArray());
-        } else {
-            plaintext = content.toInputStream();
-        }
+        InputStream plaintext = toInputStream("JWE Unencoded Payload", content);
```

Comment:
And on this line `JWE Payload` would be the appropriate name because there is no `b64`/unencoded option for JWE.

Sorry for the confusion!

Reply:
Done. And no problem. :)

Reply:
Looks good! Thanks! 🚀